common/src/main/java/org/opensearch/ml/common/ToolMetadata.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.ml.common;
+
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+
+public class ToolMetadata implements ToXContentObject, Writeable {
+
+    public static final String TOOL_NAME_FIELD = "name";
+    public static final String TOOL_DESCRIPTION_FIELD = "description";
+
+    @Getter
+    private String name;
```

Comment:
Jing did a refactor and changed name to type([code link](https://github.com/opensearch-project/ml-commons/commit/90d9b310811c682f55ee5415d53ad0be8f671156#diff-f3f27d4e953486c2798c6edef8641779b3211dabbe5374c5bfe1737b3aab2f25)). Change to type?

Reply:
Jing's change still kept `name`, just added a new field `type`. Do we want to include `type` in ToolMetadata and show it in GetTool and ListTools response?

Reply:
I think we should show `type` & `version` both.

Reply:
Yeah, we need both type and name.

Reply:
Should we add a default version each tool? Currently it's set to null - https://github.com/opensearch-project/ml-commons/blob/8352e69ae4edd2b615933ff2947d61d1d9546740/ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/AgentTool.java#L78

Reply:
Added type and version information in the latest revision